service/history/execution/mutable_state_builder.go

```diff
@@ -1982,7 +1982,7 @@ func (e *mutableStateBuilder) addBinaryCheckSumIfNotExists(
 		}
 	}
 
-	if len(currResetPoints) == maxResetPoints {
+	if len(currResetPoints) >= maxResetPoints {
```

Comment:
The 2 lines below assumes size of `currResetPoints` equals to `maxResetPoints` and discards first item. If it is possible that size is greater than max points then let's update trim offset accordingly

Reply:
and would be great to add a unit test for this function as a follow up

Reply:
Sure, I will the test. As for trimming the size, there are some existing workflows with huge list, I think it is a good idea to remove them, just not sure the safety. @davidporter-id-au Any concerns here?

Reply:
I don't see any problems in trimming this. I think we should check with the team, but I'd suggest just keeping the last 5 seems reasonable

Reply:
sorry, reading this properly for the first time. 

Yeah++ to testing and maybe a small refactoring, 

I think actually we probably want to do it now, I'm sorry to say Neil.

This whole function is a bit of a mess. It should just be a simple pure function, rather than a method,

Reply:
Sure, sounds like a good idea to refactor and move it to a func, also easier for test.

Reply:
I moved the trim part to a func and added tests. Let me know if we need to refactor more there